satpy/tests/reader_tests/utils.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities for reader tests."""
+
+
+def default_attr_processor(root, attr):
+    """Do not change the attribute."""
+    return attr
+
+
+def fill_h5(root, contents, attr_processor=default_attr_processor):
```

Comment:
Was this just cleanup of tests or was this actually required to make things work?

Reply:
No that was just cleanup. The actual problem was re-using the `var` variable in the inner loop, so I renamed it to `attr_val`.